### ISSUE TITLE
Ratio rounding results in 1.0 value before fully seeded

### ISSUE THREAD

**Issue by ruuk** [erickok/transdroid#257]
At least with Deluge, ratios seem to be rounded, which means that a torrent will show a 1.0 ratio when it has not quite reached a 1.0 ratio. This resulted in me pausing torrents before they were fully seeded.
Perhaps this could be rounded down?

Thanks

**Comment by erickok** [comment]
Well if I would round down 0.999, for example, to 0.9 or 0.999 to 0.0, that would not be accurate at all. Instead why not rely on rules to stop seeding when you are at 1.0? You can set this up in Deluge very easily.

**Comment by ruuk** [comment]
My point is that showing 1.0 on a ratio that is less than 1 seems counter intuitive to me. If you round to 1.0 you still have the same 0.1 degree of inaccuracy you would have if you round it down. The displayed ratio is not there for accuracy, and I think that it give less information as it is, especially in this context. Right now 1.0 means (I assume) ~0.95 to ~1.04 when what I want to know (and my premise is that most users want to know) is if it is yet greater or equal to 1.

In short, to me 1.0 = seeded, but it doesn't in this app.

I know how to auto pause torrents, my example was probably not the best :)

In any case, this is just my thinking. It's your app and you should of course do whatever makes sense to you :)

**Comment by erickok** [comment]
I have decided against it. While I see your point and understand the use case, the are three reasons against it:

- The ratio would be artificially adjusted. If we forcefully round down any < 1.0 then it looks like it is 'stuck' until 1.0 is reached.
- The calculation would be incorrect. For a 10GB torrent with 9.9GB upload it would be weird to display a ratio of 0.9
- The ratio displayed would be different than that in the torrent client itself.

Sorry.
